### PR TITLE
Add Custom Privacy Text Option to CTAs

### PR DESCRIPTION
**Pull request by robdivincenzo**
# Description

This PR adds a field to CTAs to overwrite the default privacy notice. This new field will allow users to A/B test custom privacy text vs. the default privacy text on various signups or petitions.

When this field is set, it will display instead of the default privacy notice. When this field is empty (as by default), the default privacy text will display.

Link to sample test page: 
Related PRs/issues: #12169 / TP1-394

Credentials:
username: admin
password: W5YO6CFsu0$98!ch

# To Test
1) Log in to review app via https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/cms/ (see credentials above)
2) Navigate to https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/en/campaigns/century-interview-plan/
3) Note the default privacy text
4) Edit this page
5) Click on the `...` actions to edit the Cta
6) Note the new field for Privacy Notice and the help text
7) Add in a new Privacy Notice, and be sure to include a link within the richtext field by highlighting some text.
8) Review the campaign page again: https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/en/campaigns/century-interview-plan/
9) Note the custom Privacy Notice is now showing

# Checklist

**Tests**
~~- [ ] Is the code I'm adding covered by tests?~~

**Changes in Models:**
~~- [ ] Did I update or add new fake data?~~
~~- [ ] Did I squash my migration?~~
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
~~- [ ] Is my code documented?~~
~~- [ ] Did I update the READMEs or wagtail documentation?~~

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-610)

### CODE REVIEW

**Comment by mmmavis** [comment]
Sorry for the delay. I will review this PR today.

**Review by mmmavis — CHANGES_REQUESTED** [review]
Yay we are almost ready to enable A/B test for newsletter signup form! I've left some inline comments, please let me know what you think! After searching for all instances of `newsletter-signup-module`, I've also noticed there are a few instances missing the `data-cta-privacy-notice` attribute. For example, in `bannered_campaign_page.html` around line 65, `network-api/networkapi/templates/fragments/primary_nav.html` around line 85, and maybe some more. Could we please add the attribute to those places? Thanks!

File: source/js/components/newsletter-signup/organisms/default-layout-signup.jsx [hunk 1]
```diff
@@ -365,6 +387,7 @@ DefaultSignupForm.propTypes = {
   ctaHeader: PropTypes.string,
   ctaDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
     .isRequired,
+  ctaHeader: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (robdivincenzo) [reply source/js/components/newsletter-signup/organisms/default-layout-signup.jsx]: De-duped! 😅

File: source/js/components/newsletter-signup/organisms/default-layout-signup.jsx [hunk 2]
```diff
@@ -156,6 +157,23 @@ class DefaultSignupForm extends Component {
     );
   }
 
+  renderPrivacyNotice() {
+    //[TODO] Investigate removing the legacy richtext template which renders an empty rich-text div wrapper
```
[the remark this answers is not in the document]
Comment (robdivincenzo) [reply source/js/components/newsletter-signup/organisms/default-layout-signup.jsx]: Sounds good to me, I've moved the logic 👍

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_body.html [hunk 3]
```diff
@@ -318,7 +318,7 @@
...
+                <label id="tfa_492-L" class="label preField" aria-label="Privacy Agreement  required">Privacy Agreement</label><br><div class="inputWrapper"><span id="tfa_492" class="choices vertical required" aria-required="true"><span class="oneChoice"><input type="checkbox" value="tfa_493" class="calc-DefaultName" id="tfa_493" name="tfa_493" aria-labelledby="tfa_493-L" data-tfa-labelledby="tfa_492-L tfa_493-L"><label class="label reqMark postField [&_*]:tw-inline" id="tfa_493-L" for="tfa_493"><span class="input-checkbox-faux [&_:before]:tw-inline-block"></span>{% if privacy_notice %}{% blocktrans with privacy_notice=privacy_notice|richtext %}{{ privacy_notice }}{% endblocktrans %}{% else %}{% blocktrans with link="https://www.mozilla.org/privacy/websites/" %}I’m okay with Mozilla handling my info as explained in this <a href="{{ link }}" target="_blank">Privacy Notice</a>.{% endblocktrans %}{% endif %}</label></span></span></div>
```
[the remark this answers is not in the document]
Comment (robdivincenzo) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_body.html]: Oh wow, that's good to know. Looks like there's a bit of work to make all of this work again when there's a new version. Moved the rules to the override!

**Comment by robdivincenzo** [comment]
Thanks for the review and feedback @mmmavis! PR should be up-to-date.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_body.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_body.html]: Yea. It's a lot of manual work involved each time. 😬 🙈

**Review by mmmavis — APPROVED** [review]
@robdivincenzo R+! Awesome work! Before you merge this PR, could you please add a note at the top of `wagtailpages/fragments/formassembly_body.html` about the changes you applied to the privacy field?

**Comment by data-sync-user** [comment]
➤ Simon Acosta Torres commented:

PR has been merged.

**Pull request merged**